lib/awful/ewmh.lua

```diff
@@ -280,7 +284,18 @@ function ewmh.geometry(c, context, hints)
             props.honor_padding = beautiful.maximized_honor_padding ~= false
         end
 
+        if original_context == "fullscreen" and beautiful.fullscreen_hide_border ~= false then
+            props.ignore_border_width = true
+        end
+
         aplace[context](c, props)
+
+        -- Remove the border to get a "real" fullscreen.
+        if original_context == "fullscreen" and beautiful.fullscreen_hide_border ~= false then
+            repair_geometry_lock = true
+            c.border_width = 0
+            repair_geometry_lock = false
```

Comment:
shouldn't this be:
```
local original = repair_geometry_lock
repair_geometry_lock = true
c.border_width = 0
repair_geometry_lock = original
```
After all, I think that this function here will be called by `repair_geometry()`.